### ISSUE TITLE
Linting for Hubot

### ISSUE THREAD

**Issue by michaelansel** [github/hubot#872]
Now that we have automated tests, we should also have a way to automatically enforce [CoffeeScript style](https://github.com/polarmobile/coffeescript-style-guide). [CoffeeLint](http://www.coffeelint.org/) looks promising.
- [ ] Fix all linter violations in the codebase
- [ ] Auto-run the linter via Travis for new PRs

**Comment by technicalpickles** [comment]
We've started using coffeelint for our internal hubot, and it's working well :+1:

The pattern we've used there and other places is to add coffeelint, and start with coffeelint.json that is pretty lenient. That can feasibly save having to make a ton of changes up front. It'd also be less disruptive to PRs that are in flight :grin: If that works out, it's easy to do follow up PRs to make it stricter and do targeted fixes.

cc @aroben
